implementations/micrometer-registry-stackdriver/src/main/java/io/micrometer/stackdriver/StackdriverMeterRegistry.java

```diff
@@ -325,11 +340,23 @@ TimeSeries createTimeSeries(Meter meter, Distribution distribution) {
 
         private TimeSeries createTimeSeries(Meter.Id id, TypedValue typedValue, MetricDescriptor.ValueType valueType,
                                             @Nullable String statistic) {
+            return createTimeSeries(id, typedValue, valueType, statistic, MetricDescriptor.MetricKind.GAUGE);
+        }
+
+        private TimeSeries createTimeSeries(Meter.Id id, TypedValue typedValue, MetricDescriptor.ValueType valueType,
+                                            @Nullable String statistic, MetricDescriptor.MetricKind metricKind) {
+            MetricDescriptor.MetricKind backendMetricKind = metricKind;
             if (client != null)
-                createMetricDescriptorIfNecessary(client, id, valueType, statistic);
+                backendMetricKind = createMetricDescriptorIfNecessary(client, id, valueType, statistic, metricKind);
 
             String metricType = metricType(id, statistic);
 
+            if (!metricKind.equals(backendMetricKind)) {
+                logger.info("MetricKind configured in Stackdriver ({}) is not matching the one expected by Micrometer for that kind of Meter ({})." +
```

Comment:
Will this be logged every publish for every time series that is mismatched? That sounds overly verbose if so. Ideally it would be logged once on the first publish and then not again. Rather than a log per time series, one per publish would also be better from a verbosity perspective.

Reply:
@shakuzen you are right it can be very verbose for the user. But this log is important. And we can't move it to the publish() method level without adding a lot of complexity, which is not justified in this case. 
So I changed the logging level to 'trace'.